assets/js/components/ClusterDetails/ClusterDetailsPage.jsx

```diff
@@ -1,32 +1,50 @@
 import React, { useEffect } from 'react';
 import { useNavigate, useParams } from 'react-router-dom';
 import { useSelector, useDispatch } from 'react-redux';
+import { get } from 'lodash';
 
 import {
   getCluster,
   getClusterHosts,
   getClusterSapSystems,
 } from '@state/selectors/cluster';
+import { getCatalog } from '@state/selectors/catalog';
+import { getLastExecution } from '@state/selectors/lastExecutions';
+import { updateCatalog } from '@state/actions/catalog';
 import {
   updateLastExecution,
   executionRequested,
 } from '@state/actions/lastExecutions';
-import { getLastExecution } from '@state/selectors/lastExecutions';
+import { TARGET_CLUSTER } from '@lib/model';
 import AscsErsClusterDetails from './AscsErsClusterDetails';
 import HanaClusterDetails from './HanaClusterDetails';
 import { getClusterName } from '../ClusterLink';
 
 export function ClusterDetailsPage() {
-  const { clusterID } = useParams();
   const navigate = useNavigate();
+  const dispatch = useDispatch();
+
+  const { clusterID } = useParams();
 
   const cluster = useSelector(getCluster(clusterID));
 
-  const dispatch = useDispatch();
+  const provider = get(cluster, 'provider');
+  const type = get(cluster, 'type');
+
+  const catalog = useSelector(getCatalog());
+
   const lastExecution = useSelector(getLastExecution(clusterID));
+
   useEffect(() => {
+    dispatch(
```

Comment:
On page load (ie refresh when in a cluster detail) there are two pairs of requests made against the catalog and the last execution.

First we have a GET /api/v1/checks/catalog?target_type=cluster and then a complete GET /api/v1/checks/catalog?provider=azure&target_type=cluster&cluster_type=hana_scale_up

I guess we should check whether we have everything we need in order to dispatch an update of the catalog, and then dispatch it.

We could have the `cluster` variable as a dependency of this useEffect and if it is not null/undefined, then it means we have everything in place and then trigger the requests. What do you think?

Reply:
I told him to write that `useEffect` that way :)

thing is, putting an object in useEffect dependencies can trigger unwated side effects because every time the object is replaced it gets a new reference and can lead to additional calls made. This way otherwise we make the right call when things change punctually. I personally thought we can live with an extra API call :+1:

Reply:
I guess there can be a middle ground where we do not use the object (cluster), but we check on what we are passing (provider and type) and make the call only of we have both of them.

Would that work?